Chandra/cmd_states/cmd_states.py

```diff
@@ -393,14 +399,16 @@ def get_state0(date=None, db=None, date_margin=10, datepar='datestop'):
         db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read',
                          database='aca')
 
-    # Date for which cmd_states are certainly reliable
-    definitive_date = DateTime(time.time() - date_margin * 86400.,
-                               format='unix').date
-    if date is None or date > definitive_date:
-        date = definitive_date
-    else:
+    if date is not None:
```

Comment:
But it wasn't exactly a conversation.  It looks to me like you've taken half of my logic here, modified it for a `None` in `date_margin` (for reasons that I don't think were explained anyway) and then used the rest of the logic to define requirements on `date` that don't need to exist.  I still think the block could look like:
```
    date = DateTime(date).date

    if date_margin is not None:
        # Date for which cmd_states are certainly reliable
        definitive_date = DateTime(-1 * date_margin).date
        if date > definitive_date:
            date = definitive_date

    state0 = db.fetchone("""SELECT * FROM cmd_states
                            WHERE %s < '%s'
                            AND pcad_mode = 'NPNT'
                            ORDER BY %s DESC""" % (datepar, date, datepar))
```